server/server.go

```diff
@@ -86,22 +85,23 @@ func NewShipperServer(cfg Config, publisher Publisher) (ShipperServer, error) {
 	return &s, nil
 }
 
-// GetAcceptedIndex returns the accepted index
-func (serv *shipperServer) GetAcceptedIndex() uint64 {
-	return uint64(serv.publisher.AcceptedIndex())
-}
-
-// GetPersistedIndex returns the persisted index
-func (serv *shipperServer) GetPersistedIndex() uint64 {
-	return uint64(serv.publisher.PersistedIndex())
+// getPersistedIndex returns the persisted index, or 0 if there is an error.
+// (An error can only happen if the queue is closed, in which case a placeholder
+// of 0 is often sufficient. Callers that need to explicitly recognize an error
+// state should use publisher.PersistedIndex directly.)
+func (serv *shipperServer) getPersistedIndex() uint64 {
+	index, err := serv.publisher.PersistedIndex()
+	if err != nil {
+		// An error means the queue has been closed, so just return a placeholder.
+		return 0
+	}
+	return uint64(index)
 }
```

Comment:
I believe we don't need this helper after we removed the PersistedIndex field from the PublishReply type.

```suggestion
```

Reply:
done